main.go

```diff
@@ -63,10 +64,14 @@ var (
 	// counters
 	colorCounters     = simulation.NewColorCounters()
 	adversaryCounters = simulation.NewColorCounters()
+	nodeCounters      = []*simulation.AtomicCounters{}
```

Comment:
Do not use slice of pointers. I would suggest changing the type returned by  `simulation.NewAtomicCounters()` and `simulation.NewColorCounters()` to be non-pointer type, because I don't see a need for it to be a pointer.